packages/babel-parser/src/tokenizer/index.js

```diff
@@ -498,6 +498,14 @@ export default class Tokenizer extends ParserErrors {
     }
   }
 
+  replaceToken(type: TokenType): void {
+    this.state.type = type;
+    // the prevType of updateContext is required
+    // only when the new type is tt.slash/tt.jsxTagEnd
+    // $FlowIgnore
```

Comment:
We might consider marking it as optional?

Reply:
I am working on completely removing `updateContext()` in base parser and jsx parser, at least currently we don't need `updateContext` in the base parser so hopefully `replaceToken` can be moved to the jsx plugin.